payjoin/src/core/receive/v2/session.rs

```diff
@@ -597,88 +563,88 @@ mod tests {
 
-        Ok(())
+        let test = SessionHistoryTest {
+            events,
+            expected_session_history: SessionHistoryExpectedOutcome {
+                fallback_tx: None,
+                expected_status: SessionStatus::Failed,
+            },
+            expected_receiver_state: ReceiveSession::HasReplyableError(Receiver {
+                state: HasReplyableError { error_reply: (&expected_error).into() },
+                session_context: SessionContext { reply_key, ..session_context },
+            }),
+        };
+        run_session_history_test(test)
     }
 
     #[test]
-    fn test_skipped_session_extract_err_request() -> Result<(), BoxError> {
-        let ohttp_relay = EXAMPLE_URL;
-        let mock_err = mock_err();
+    fn test_session_transient_error() -> Result<(), BoxError> {
+        let persister = NoopSessionPersister::<SessionEvent>::default();
+        let session_context = SHARED_CONTEXT.clone();
+        let mut events = vec![];
 
-        let session_history =
-            SessionHistory { events: vec![SessionEvent::CheckedBroadcastSuitability()] };
-        let err_req = session_history.extract_err_req(ohttp_relay)?;
-        assert!(err_req.is_none());
+        let original = original_from_test_vector();
+        // Mock some implementation error
+        let _maybe_broadcastable = unchecked_receiver_from_test_vector()
+            .check_broadcast_suitability(None, |_| Err("mock error".into()))
+            .save(&persister)
+            .expect_err("Mock error should error");
+        // NOTE: it would be good to assert against the internal error type but InternalPersistedError is private
 
-        let session_history = SessionHistory {
-            events: vec![
-                SessionEvent::CheckedBroadcastSuitability(),
-                SessionEvent::SessionInvalid(mock_err.0.clone(), Some(mock_err.1.clone())),
-            ],
-        };
+        let reply_key = Some(crate::HpkeKeyPair::gen_keypair().1);
 
-        let err_req = session_history.extract_err_req(ohttp_relay)?;
-        assert!(err_req.is_none());
+        events.push(SessionEvent::Created(session_context.clone()));
+        events.push(SessionEvent::RetrievedOriginalPayload {
+            original: original.clone(),
+            reply_key: reply_key.clone(),
+        });
 
-        let session_history = SessionHistory {
-            events: vec![
-                SessionEvent::Created(SHARED_CONTEXT.clone()),
-                SessionEvent::CheckedBroadcastSuitability(),
-                SessionEvent::SessionInvalid(mock_err.0.clone(), Some(mock_err.1.clone())),
-            ],
+        let test = SessionHistoryTest {
+            events,
+            expected_session_history: SessionHistoryExpectedOutcome {
+                fallback_tx: None,
+                expected_status: SessionStatus::Active,
+            },
+            expected_receiver_state: ReceiveSession::UncheckedOriginalPayload(Receiver {
```

Comment:
Should the expected typestate be `HasReplyError`? Perhaps not bc the error is transient. ~In that case should we have a seperate unit test that returns `Ok(true)` in the closure and checks if we are in the has reply error state?~ 
Edit: i see this is already unit tested in `receive/v2/mod.rs`. Thanks